modules/storage-api/src/test/java/org/apache/ignite/internal/storage/BaseMvStoragesTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage;
+
+import java.util.Locale;
+import java.util.Objects;
+import org.apache.ignite.internal.schema.BinaryRow;
+import org.apache.ignite.internal.schema.Column;
+import org.apache.ignite.internal.schema.NativeTypes;
+import org.apache.ignite.internal.schema.SchemaDescriptor;
+import org.apache.ignite.internal.schema.marshaller.KvMarshaller;
+import org.apache.ignite.internal.schema.marshaller.MarshallerException;
+import org.apache.ignite.internal.schema.marshaller.MarshallerFactory;
+import org.apache.ignite.internal.schema.marshaller.reflection.ReflectionMarshallerFactory;
+import org.apache.ignite.internal.schema.row.Row;
+import org.apache.ignite.lang.IgniteException;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+/**
+ * Base test for MV storages, contains pojo classes, their descriptor and a marshaller instance.
+ */
+public class BaseMvStoragesTest {
+    /** Default reflection marshaller factory. */
+    protected static MarshallerFactory marshallerFactory;
```

Comment:
Here's my reasoning:
- these fields are the same for every test case, that's why I made them static
- initialization is in BeforeAll for consistency with AfterAll
- fields are not final because I clean them afterwards to avoid any potential memory leaks

Reply:
I think memory leaks in tests are usually not something that we should be concerned about :) Also, if the fields are not static, but instance fields, their values are garbage as soon as the tests are finished in the given test class, so nullifying the fields seems to be unnecessary, hence `@AfterAll` goes away, etc.

Reply:
I see. In Ignite 2.x we had several major leaks in a very similar scenario, so for me it's a good habit to nullify static fields when tests don't need them anymore